### PR TITLE
plausible: 1.3.0 -> 1.4.0

### PR DESCRIPTION
**Pull request by happysalada**
###### Motivation for this change

upstream update.

I have one question. I was thinking of using the recently made `fetchYarnDeps` to not need the giant yarn.nix file. This would require a manual change to the upstream package.json. Basically package.json has two dependencies using the version format `file:path`, and those wouldn't work with fetchYarnDeps. At the moment the upstream package.json version for those 2 files needs to be amended anyways (to change the path). I am considering just fixing the version (which can be found in the mix.lock file). The drawback is that this is a manual change that would be hard to automate. The benefit is that it enables the use of fetchYarnDeps. Happy to have feedback on this.

@yu-re-ka btw tried using fetchYarnDeps with an empty hash, and it returned an error. It would be cool if like other fetchers, if you supply an empty hash, it assumes the `lib.fakeSha256`.

This PR only updates the package, there are also changes to the modules that I am currently testing.

(detail, but I opened a PR upstream to add the version to the package.json to not have problem with yarn https://github.com/plausible/analytics/pull/1416).
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by happysalada** [comment]
hmmm, everything is working, however the admin user password seems to be different from the one supplied.
I want to debug but don't know how to get an environment with all the same environment variables and a tty.
It seems I should use `systemd-run`, if anybody knows how to use it, I am interested.
with elixir I can get a remote console into the running program to check what went wrong. I just don't know how to use systemd-run. I'll have a look in more details tomorrow.

**Comment by happysalada** [comment]
@Ma27 this is ready for review. I was going insane, I just had a newline in my password file. Well live and learn.
No rush on this, let me know if you have any feedback.

**Comment by happysalada** [comment]
I think I have found a nice way to provide users with a way to connect to a running plausible node to do some debugging, however, I'll try that in a subsequent PR (will test the change to fetchYarnDeps as well if you are open to it.)

File: nixos/modules/services/web-apps/plausible.nix [hunk 1]
```diff
@@ -231,49 +222,52 @@ in {
...
+            ] ++ lib.optionals (cfg.mail.smtp.passwordFile != null) [ "SMTP_USER_PWD:${cfg.mail.smtp.passwordFile}"];
           };
         };
       }
       (mkIf cfg.database.postgres.setup {
         # `plausible' requires the `citext'-extension.
         plausible-postgres = {
           after = [ "postgresql.service" ];
-          bindsTo = [ "postgresql.service" ];
-          requiredBy = [ "plausible.service" ];
```
Comment (Ma27) [inline nixos/modules/services/web-apps/plausible.nix:264]: Hmm, whyis this not needed anymore?

File: pkgs/servers/web-apps/plausible/default.nix [hunk 2]
```diff
@@ -5,43 +5,35 @@
...
-
-    # We need ecto 3.6 as this version checks whether the database exists before
-    # trying to create it. The creation attempt would always require super-user privileges
-    # and since 3.6 this isn't the case anymore.
-    patches = [
-      ./ecto_sql-fix.patch
-      ./plausible-Bump-clickhouse_ecto-dependency-to-be-compatible-with-ecto-3.6.patch
-    ];
+    sha256 = "sha256-ZQfrTxsLzCWFf3vabOk0vyHWZLw69GJovm3vR+7UbMY=";
   };
```
Comment (Ma27) [inline pkgs/servers/web-apps/plausible/default.nix:29]: Does this still work fine with the update script?

File: nixos/modules/services/web-apps/plausible.nix (hunk 1 above)
Comment (happysalada) [reply nixos/modules/services/web-apps/plausible.nix:264]: My bad for not detailing a bit more.
bindsTo is just a strong form of requires, where if the required dependency stops, then the dependency stops as well. It doesn't really make sense for a setup script that just runs for several seconds (IMO). Open to discussion of course.
requiredBy, is usually never used in the configuration settings directly and is just the symetric of required.
https://www.freedesktop.org/software/systemd/man/systemd.unit.html#Mapping%20of%20unit%20properties%20to%20their%20inverses
(from the Note section).
Perhaps I read that in the wrong way. Open for discussions as well.

File: pkgs/servers/web-apps/plausible/default.nix (hunk 2 above)
Comment (happysalada) [reply pkgs/servers/web-apps/plausible/default.nix:29]: Just updated the script to make sure it works with the new version.

**Review by Ma27 — CHANGES_REQUESTED** [review]
The test doesn't evaluate anymore. First of all, you have to use `runCommand` rather than `runCommandNoCC` (fixed in a `yarn2nix` on nixpkgs master IIRC) and when you do this, it still fails like this:

```
error: undefined variable 'plausibleConsole'

       at /home/ma27/Projects/nixpkgs/nixos/modules/services/web-apps/plausible.nix:166:36:

          165|
          166|     environment.systemPackages = [ plausibleConsole ];
             |                                    ^
          167|
(use '--show-trace' to show detailed location information)

```

**Comment by happysalada** [comment]
@Ma27 My bad about missing those. I've committed fixes.

**Comment by yu-re-ka** [comment]
Can we put the package changes into a seperate PR, so I can review and merge them independently of the ongoing discussions regarding module changes?

**Comment by happysalada** [comment]
I would like to, but there are some updates needed to the module as well to make the new version work. Some environment variable names that changed. I realise though that this PR was a bit large, and should have been broken down a little more before hand.

**Comment by happysalada** [comment]
I'll tell you what, if your PR is ready before this is approved, then just merge it, I'll do the rebase afterwards, no worries.

**Comment by Ma27** [comment]
It's on my todo list to review & merge this week. If that's good enough for @yu-re-ka, then we can leave it as-is :)

**Comment by yu-re-ka** [comment]
> I would like to, but there are some updates needed to the module as well to make the new version work

I see, then it has to be one big PR, which is also fine.

> It's on my todo list to review & merge this week

Definitely good enough :)
I'm also running the tests now so maybe you will have less work once you get around to reviewing it.

**Comment by yu-re-ka** [comment]
✅  Package builds, test passes on x86_64-linux

**Review by Ma27 — APPROVED** [review]

**Comment by nh2** [comment]
> The problem with the epmd started by plausible is that when you restart the service, it becomes a zombie process.

@happysalada Related question: https://github.com/NixOS/nixpkgs/pull/130297#issuecomment-1011743934

**Comment by nh2** [comment]
@happysalada That doesn't quite answer it -- I'll ask in the linked thread.

**Comment by RaitoBezarius** [comment]
This broke secret handling for Plausible and it seems this was never tested/debugged properly…

`LoadCredential` do not create any environment variables and require read from the filesystem.

**Pull request merged**